ruby/tests/well_known_types_test.rb

```diff
@@ -139,4 +139,31 @@ def test_any
     assert any.is(Google::Protobuf::Timestamp)
     assert_equal ts, any.unpack(Google::Protobuf::Timestamp)
   end
+
+  def test_struct_assign
+    s = Google::Protobuf::Struct.new(fields: {'a' => Google::Protobuf::Value.new({number_value: 4.4})})
+    assert_equal 4.4, s['a']
+
+    s = Google::Protobuf::Struct.new(fields: {'a' => {number_value: 2.2}})
+    assert_equal 2.2, s['a']
```

Comment:
i would also like to see `Google::Protobuf::Struct#[]` convert the method argument to a String, so this would also work:

```ruby
assert_equal 2.2, s[:a]
```

Reply:
you got it. How's this look?

Reply:
Looks great!